modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v3_1_1/KaleoNotificationUpgradeProcess.java

```diff
@@ -48,16 +48,19 @@ protected void doUpgrade() throws Exception {
 			ResultSet resultSet = preparedStatement1.executeQuery();
 
 			while (resultSet.next()) {
-				String[] notificationTypes = Stream.of(
-					StringUtil.split(resultSet.getString("notificationTypes"))
-				).filter(
-					notificationType -> !Objects.equals(notificationType, "im")
-				).filter(
-					notificationType -> !Objects.equals(
-						notificationType, "private-message")
-				).toArray(
-					String[]::new
-				);
+				String[] notificationTypes = TransformUtil.transform(
+					StringUtil.split(resultSet.getString("notificationTypes")),
+					notificationType -> {
+						if (Objects.equals("im", notificationType) ||
+							Objects.equals(
+								notificationType, "private-message")) {
```

Comment:
@ling-alan-huang Here we try to put the looped item `notificationType` the second operand in the equals comparison, but SF automatically changes the one in line 55~56 back. Could you take a look why it's not enforcing the first one?

Reply:
I will have a look at it tomorrow.

Reply:
@qz1010 Can you have a look at it? I am off today.